pkgs/by-name/fo/foks-server/package.nix

```
@@ -0,0 +1,90 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  buildNpmPackage,
+  pcsclite,
+  pkg-config,
+  stdenv,
+  tailwindcss_4,
+  templ,
+  uglify-js,
+}:
+
+let
+  src = fetchFromGitHub {
+    owner = "foks-proj";
+    repo = "go-foks";
+    tag = "v${version}";
+    hash = "sha256-BeDhq+963G7OA164HcBJ8njfwPhRbMxPuckk7V538WI=";
+  };
+  version = "0.1.2";
+  templFoks = templ.overrideAttrs (old: {
+    pname = "templ-foks";
+    version = "0.3.833";
+    src = old.src.override {
+      hash = "sha256-4K1MpsM3OuamXRYOllDsxxgpMRseFGviC4RJzNA7Cu8=";
+    };
+    vendorHash = "sha256-OPADot7Lkn9IBjFCfbrqs3es3F6QnWNjSOHxONjG4MM=";
+  });
+  foksNodeModules = buildNpmPackage {
+    pname = "foks-node-modules";
+    inherit version src;
+    npmDepsHash = "sha256-mASu3taprCYP+muRh4b+qUTce7YTdISf/QnmSvUy6Mo=";
+    dontNpmBuild = true;
+    sourceRoot = "${src.name}/server/web/frontend";
+    installPhase = ''
+      mkdir -p $out
+      cp -r node_modules $out/node_modules
+    '';
+  };
+in
+buildGoModule (finalAttrs: {
+  pname = "foks-server";
+  inherit version src;
+
+  vendorHash = "sha256-8/SVOWMoCfeiuH2As2cC/HLRs1WQIQ4/Ko1olXDq6bo=";
+
+  postPatch = ''
+    # Generate templates
+    cd ./server/web/templates
+    ${templFoks}/bin/templ generate
+    cd -
+    # Generate tailwind
+    cd ./server/web/frontend
+    mkdir -p ../static/css
+    ${tailwindcss_4}/bin/tailwindcss -i ./css/input.css -o ../static/css/style.min.css --minify
+    ${tailwindcss_4}/bin/tailwindcss -i ./css/input.css -o ../static/css/style.css
+    ${uglify-js}/bin/uglifyjs -c < ../static/js/foks.js > ../static/js/foks.min.js
+    cd -
+    # Copy htmx dependency from node modules
+    cd ./server/web/static
+    mkdir -p ./js
+    cp ${foksNodeModules}/node_modules/htmx.org/dist/htmx.js ./js/htmx.js
+    cp ${foksNodeModules}/node_modules/htmx.org/dist/htmx.min.js ./js/htmx.min.js
+    cd -
+  '';
```

```suggestion
  postPatch =
  # Generate templates
  ''
    pushd ./server/web/templates
    ${templFoks}/bin/templ generate
    popd
  ''
  # Generate tailwind
  + ''
    pushd ./server/web/frontend
    mkdir -p ../static/css
    ${lib.getExe tailwindcss_4} -i ./css/input.css -o ../static/css/style.min.css --minify
    ${lib.getExe tailwindcss_4} -i ./css/input.css -o ../static/css/style.css
    ${lib.getExe uglify-js} -c < ../static/js/foks.js > ../static/js/foks.min.js
    popd
  ''
  # Copy htmx dependency from node modules
  + ''
    pushd ./server/web/static
    mkdir -p ./js
    cp ${foksNodeModules}/node_modules/htmx.org/dist/htmx.js ./js/htmx.js
    cp ${foksNodeModules}/node_modules/htmx.org/dist/htmx.min.js ./js/htmx.min.js
    popd
  '';
```

Nits (non-blocking).
`lib.getExe` is preferred from hardcoding `/bin/foo`. Also, prefer leaving comments outside of nix strings, so that they can be changed without triggering any rebuild.

---

Good nits. Ty